pymongo/asynchronous/client_session.py

```diff
@@ -1129,33 +1132,30 @@ def get_server_session(self, session_timeout_minutes: Optional[int]) -> _ServerS
         self._clear_stale(session_timeout_minutes)
 
         # The most recently used sessions are on the left.
-        while self:
-            s = self.popleft()
+        while True:
+            try:
+                s = self.popleft()
+            except IndexError:
+                break
             if not s.timed_out(session_timeout_minutes):
                 return s
 
         return _ServerSession(self.generation)
 
-    def return_server_session(
-        self, server_session: _ServerSession, session_timeout_minutes: Optional[int]
```

Comment:
Because session_timeout_minutes was only used to clear stale sessions and it seems redundant to do that on both get_session and return_session. It's good enough to clear stale sessions only when getting a session from the pool.